gempython/tools/amc_user_functions_xhal.py

```diff
@@ -473,34 +487,54 @@ def getTriggerLinkStatus(self,printSummary=False, checkCSCTrigLink=False, ohMask
+            lines[0].append("{}".format(" "*(len(max(ohlinkfmt,key=len)))))
+            for st in range(len(ohlinkfmt)):
+                lines[st+1].append("{{:{}s}}".format(len(max(ohlinkfmt,key=len))).format(ohlinkfmt[st]))
+
+            if (checkCSCTrigLink):
+                clines = [[] for x in range(len(ohlinkfmt)+1)]
+                clines[0].append("{}".format(" "*(len(max(ohlinkfmt,key=len)))))
+                for st in range(len(ohlinkfmt)):
+                    clines[st+1].append("{{:{}s}}".format(len(max(ohlinkfmt,key=len))).format(ohlinkfmt[st]))
+
         for ohN in range(self.nOHs):
             # Skip Masked OH's
             if (not ((ohMask >> ohN) & 0x1)):
                 continue
 
-            if(printSummary):
-                print("----------OH{0}----------".format(ohN))
-                print("LINK0_MISSED_COMMA_CNT		{0}0x{1:x}{2}".format(colors.RED if (linkResult[ohN] > 0) else colors.GREEN,linkResult[ohN],colors.ENDC))
-                print("LINK1_MISSED_COMMA_CNT		{0}0x{1:x}{2}".format(colors.RED if (linkResult[ohN+self.nOHs] > 0) else colors.GREEN,linkResult[ohN+self.nOHs],colors.ENDC))
-                print("LINK0_OVERFLOW_CNT		{0}0x{1:x}{2}".format(colors.RED if (linkResult[ohN+2*self.nOHs] > 0) else colors.GREEN,linkResult[ohN+2*self.nOHs],colors.ENDC))
-                print("LINK1_OVERFLOW_CNT		{0}0x{1:x}{2}".format(colors.RED if (linkResult[ohN+3*self.nOHs] > 0) else colors.GREEN,linkResult[ohN+3*self.nOHs],colors.ENDC))
-                print("LINK0_UNDERFLOW_CNT		{0}0x{1:x}{2}".format(colors.RED if (linkResult[ohN+4*self.nOHs] > 0) else colors.GREEN,linkResult[ohN+4*self.nOHs],colors.ENDC))
-                print("LINK1_UNDERFLOW_CNT		{0}0x{1:x}{2}".format(colors.RED if (linkResult[ohN+5*self.nOHs] > 0) else colors.GREEN,linkResult[ohN+5*self.nOHs],colors.ENDC))
-                print("LINK0_SBIT_OVERFLOW_CNT		{0}0x{1:x}{2}".format(colors.RED if (linkResult[ohN+6*self.nOHs] > 0) else colors.GREEN,linkResult[ohN+6*self.nOHs],colors.ENDC))
-                print("LINK1_SBIT_OVERFLOW_CNT		{0}0x{1:x}{2}".format(colors.RED if (linkResult[ohN+7*self.nOHs] > 0) else colors.GREEN,linkResult[ohN+7*self.nOHs],colors.ENDC))
-                pass
+            if printSummary:
+                lines[0].append(hfmt.format(("OH{:d}".format(ohN)).rjust(4)))
+                for st in range(len(ohlinkfmt)):
+                    if st < 1:
+                        lines[st+1].append(hfmt.format(stfmt.format(colors.RED if not (linkResult[ohN]) else colors.GREEN, linkResult[ohN],  colors.ENDC)))
+                    elif st < 2:
+                        lines[st+1].append(hfmt.format(stfmt.format(colors.RED if not (linkResult[ohN+self.nOHs]) else colors.GREEN, linkResult[ohN+self.nOHs],  colors.ENDC)))
+                    else:
+                        lines[st+1].append(hfmt.format(stfmt.format(colors.RED if not (linkResult[ohN+(st*self.nOHs)]) else colors.GREEN, linkResult[ohN+(st*self.nOHs)],  colors.ENDC)))
```

Comment:
This didn't seem to work as intended; when tried on QC8 the numbers came out as red even though the counters all had values of `0x0`.